association.go

```diff
@@ -2698,13 +2705,21 @@ func (a *Association) onRetransmissionFailure(id int) {
 
 	if id == timerT1Init {
 		a.log.Errorf("[%s] retransmission failure: T1-init", a.name)
-		a.handshakeCompletedCh <- ErrHandshakeInitAck
+		select {
+		case a.handshakeCompletedCh <- ErrHandshakeInitAck:
+		case <-a.closeWriteLoopCh: // check the read/write sides
+		case <-a.readLoopCloseCh:
+		}
 		return
 	}
 
 	if id == timerT1Cookie {
 		a.log.Errorf("[%s] retransmission failure: T1-cookie", a.name)
-		a.handshakeCompletedCh <- ErrHandshakeCookieEcho
+		select {
+		case a.handshakeCompletedCh <- ErrHandshakeCookieEcho:
+		case <-a.closeWriteLoopCh: // check the read/write sides
+		case <-a.readLoopCloseCh:
```

Comment:
Should we factor this out into 
`func (a *Association) completeHandshake(err error) error`

Reply:
I'm not sure given that not all the selects contain the same cases. Although making them the same cases probably can't hurt?

Reply:
I am fine either way.

Reply:
Given they are receives, I'll refactor it :)